### PR TITLE
[ONNX] Add Internal Utils: onnx_proto_utils.py for onnx/onnx-script/onnx_proto

### PR DESCRIPTION
**Pull request by titaiwangms**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #89078
* __->__ #88376
* #86907

Added `onnx_proto_utils.py` for onnx/onnx-script related process. The idea is like jit_utils.py, and to simplify what we have in `torch/onnx/utils.py`

### CODE REVIEW

File: test/onnx/test_onnxscript_no_runtime.py [hunk 1]
```diff
@@ -92,7 +94,9 @@ def custom_layer_norm(
         y = torch.randn(N, C)
         model_layer_norm = torch.nn.LayerNorm(C)
         layer_norm_onnx = io.BytesIO()
-        torch.onnx.export(model_layer_norm, y, layer_norm_onnx, opset_version=self.opset_version)
```
Comment (AllenTiTaiWang) [inline test/onnx/test_onnxscript_no_runtime.py:95]: These lint causing by not doing lint from previous PR. Will fix this.

File: torch/onnx/_internal/onnx_proto_utils.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    # as os.PathLike[str] uncheckable at runtime
+    if export_type == _exporter_states.ExportTypes.PROTOBUF_FILE:
+        assert len(export_map) == 0
+        with torch.serialization._open_file_like(f, "wb") as opened_file:
+            opened_file.write(model_bytes)
+    elif export_type in [
+        _exporter_states.ExportTypes.ZIP_ARCHIVE,
+        _exporter_states.ExportTypes.COMPRESSED_ZIP_ARCHIVE,
+    ]:
+        compression = (
```
Comment (justinchuby) [inline torch/onnx/_internal/onnx_proto_utils.py:34]:
```suggestion
    elif export_type in {
        _exporter_states.ExportTypes.ZIP_ARCHIVE,
        _exporter_states.ExportTypes.COMPRESSED_ZIP_ARCHIVE,
    }:
```

**Review by justinchuby — APPROVED** [review]
LGTM🚀

File: torch/onnx/_internal/onnx_proto_utils.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+import io
+import os
+import zipfile
+from typing import Mapping, Union
+
+import torch
+import torch.jit._trace
+import torch.serialization
+from torch.onnx import _constants, _exporter_states, errors  # noqa: F401
```
[the remark this answers is not in the document]
Comment (AllenTiTaiWang) [reply torch/onnx/_internal/onnx_proto_utils.py]: DOne

File: torch/onnx/_internal/onnx_proto_utils.py (hunk 2 above)
Comment (AllenTiTaiWang) [reply torch/onnx/_internal/onnx_proto_utils.py]: Done

File: torch/onnx/_internal/onnx_proto_utils.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+            opened_file.write(model_bytes)
+
+        for k, v in export_map.items():
+            weight_proto_file = os.path.join(f, k)  # type: ignore[arg-type]
+            with torch.serialization._open_file_like(
+                weight_proto_file, "wb"
+            ) as opened_file:
+                opened_file.write(v)
+    else:
+        raise RuntimeError("Unknown export type")
```
[the remark this answers is not in the document]
Comment (AllenTiTaiWang) [reply torch/onnx/_internal/onnx_proto_utils.py]: Done

**Review by BowenBao — APPROVED** [review]

**Pull request closed without merging**